manager/scheduler/scheduler_test.go

```diff
@@ -982,47 +982,64 @@ func TestSchedulerPluginConstraint(t *testing.T) {
 }
 
 func BenchmarkScheduler1kNodes1kTasks(b *testing.B) {
-	benchScheduler(b, 1e3, 1e3, false)
+	benchScheduler(b, 1e3, 1e3, false, false)
 }
 
 func BenchmarkScheduler1kNodes10kTasks(b *testing.B) {
-	benchScheduler(b, 1e3, 1e4, false)
+	benchScheduler(b, 1e3, 1e4, false, false)
 }
 
 func BenchmarkScheduler1kNodes100kTasks(b *testing.B) {
-	benchScheduler(b, 1e3, 1e5, false)
+	benchScheduler(b, 1e3, 1e5, false, false)
 }
 
 func BenchmarkScheduler100kNodes100kTasks(b *testing.B) {
-	benchScheduler(b, 1e5, 1e5, false)
+	benchScheduler(b, 1e5, 1e5, false, false)
 }
 
 func BenchmarkScheduler100kNodes1MTasks(b *testing.B) {
-	benchScheduler(b, 1e5, 1e6, false)
+	benchScheduler(b, 1e5, 1e6, false, false)
+}
+
+func BenchmarkSchedulerConstraints1kNodes1kTasks(b *testing.B) {
+	benchScheduler(b, 1e3, 1e3, true, false)
+}
+
+func BenchmarkSchedulerConstraints1kNodes10kTasks(b *testing.B) {
+	benchScheduler(b, 1e3, 1e4, true, false)
+}
+
+func BenchmarkSchedulerConstraints1kNodes100kTasks(b *testing.B) {
+	benchScheduler(b, 1e3, 1e5, true, false)
+}
+
+func BenchmarkSchedulerConstraints5kNodes100kTasks(b *testing.B) {
+	benchScheduler(b, 5e3, 1e5, true, false)
 }
 
 func BenchmarkSchedulerWorstCase1kNodes1kTasks(b *testing.B) {
-	benchScheduler(b, 1e3, 1e3, true)
+	benchScheduler(b, 1e3, 1e3, false, true)
 }
 
 func BenchmarkSchedulerWorstCase1kNodes10kTasks(b *testing.B) {
-	benchScheduler(b, 1e3, 1e4, true)
+	benchScheduler(b, 1e3, 1e4, false, true)
 }
 
 func BenchmarkSchedulerWorstCase1kNodes100kTasks(b *testing.B) {
-	benchScheduler(b, 1e3, 1e5, true)
+	benchScheduler(b, 1e3, 1e5, false, true)
 }
 
 func BenchmarkSchedulerWorstCase100kNodes100kTasks(b *testing.B) {
-	benchScheduler(b, 1e5, 1e5, true)
+	benchScheduler(b, 1e5, 1e5, false, true)
 }
 
 func BenchmarkSchedulerWorstCase100kNodes1MTasks(b *testing.B) {
-	benchScheduler(b, 1e5, 1e6, true)
+	benchScheduler(b, 1e5, 1e6, false, true)
 }
 
-func benchScheduler(b *testing.B, nodes, tasks int, worstCase bool) {
+func benchScheduler(b *testing.B, nodes, tasks int, constraints, worstCase bool) {
```

Comment:
Might be better change `constraints` to `networkConstraints`. It's different from `--constraint` in service.